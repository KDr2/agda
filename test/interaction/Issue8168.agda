data ℕ : Set where
  zero : ℕ
  suc  : (n : ℕ) → ℕ

{-# BUILTIN NATURAL ℕ #-}

data Ty : Set where
  bool : Ty
  _⇒_  : (a b : Ty) → Ty

data Context : Set where
  ε : Context
  _·_ : (Γ : Context) (a : Ty) → Context

module _ where
  private
    variable
      a b : Ty
      Γ Δ : Context

  data Term : (Γ : Context) (a : Ty) → Set where
    var :                                       Term (Γ · a) a
    wk  : (e : Term Γ a)                      → Term (Γ · b) a
    app : (f : Term Γ (a ⇒ b)) (e : Term Γ a) → Term Γ b
    abs : (e : Term (Γ · a) b)                → Term Γ (a ⇒ b)

module One where

  δ : ∀ {a} → ℕ → Term ε {!  !}  -- C-c C-a
  δ zero = abs (abs (app (wk var) (app (wk var) var)))
  δ (suc n) = app (δ n) (δ zero)

  -- error: [ParseError]
  -- in the name _a_61, the part 61 is not valid because it is a literal

  -- N.B.: Solution is (a ⇒ a) ⇒ (a ⇒ a)

module Two where

  δ : ∀ {a} → ℕ → Term ε {!  !}  -- C-c C-a
  δ zero = abs (abs (app (wk var) (app (wk var) var)))
  δ (suc n) = app (δ n) (δ zero)
