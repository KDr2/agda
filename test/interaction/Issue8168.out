(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals, Errors*" "?0 : Ty ?1 : Ty Sort _47 [ at Issue8168.agda:29.10-11 ] _48 : _47 [ at Issue8168.agda:29.10-11 ] _a_56 : Ty [ at Issue8168.agda:30.17-20 ] _a_79 : _48 [ at Issue8168.agda:31.20-21 ] _a_82 : _48 [ at Issue8168.agda:31.26-27 ] Sort _85 [ at Issue8168.agda:40.10-11 ] _86 : _85 [ at Issue8168.agda:40.10-11 ] _a_94 : Ty [ at Issue8168.agda:41.17-20 ] _a_117 : _86 [ at Issue8168.agda:42.20-21 ] _a_120 : _86 [ at Issue8168.agda:42.26-27 ] ———— Error ————————————————————————————————————————————————— error: [UnsolvedConstraints] Failed to solve the following constraints: _a_94 ⇒ _a_94 = _a_94 : Ty (blocked on _a_94) _a_94 = _a_94 ⇒ _a_94 : Ty (blocked on _a_94) _a_56 ⇒ _a_56 = _a_56 : Ty (blocked on _a_56) _a_56 = _a_56 ⇒ _a_56 : Ty (blocked on _a_56)" nil)
((last . 1) . (agda2-goals-action '(0 1)))
(agda2-status-action "")
(agda2-info-action "*Auto*" "?0 := ((_a_56 ⇒ _a_56) ⇒ (_a_56 ⇒ _a_56)) (fails to parse) error: [ParseError] in the name _a_56, the part 56 is not valid because it is a literal " nil)
((last . 1) . (agda2-goals-action '(0 1)))
(agda2-status-action "")
(agda2-info-action "*Auto*" "?0 := ((_a_56 ⇒ _a_56) ⇒ (_a_56 ⇒ _a_56)) (fails to parse) error: [ParseError] in the name _a_56, the part 56 is not valid because it is a literal ?1 := ((_a_94 ⇒ _a_94) ⇒ (_a_94 ⇒ _a_94)) (fails to parse) error: [ParseError] in the name _a_94, the part 94 is not valid because it is a literal " nil)
((last . 1) . (agda2-goals-action '(0 1)))
